charts/backbeat-api/templates/deployment.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "backbeat-api.fullname" . }}
+  labels:
+    app: {{ template "backbeat-api.name" . }}
+    chart: {{ template "backbeat-api.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
```

Comment:
Could you try deploying this using `apps/v1` `apiVersion` and check whether things still work?